### PR TITLE
Enable e2e tests on helm charts

### PR DESCRIPTION
**Pull request by gianrubio**
This PR will enable e2e tests on the helm charts. It will setup kube-prometheus and prometheus-operator and wait until all the pods/containers are in ready state.

### CODE REVIEW

File: helm/hack/e2e-test.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# wait until all pods, including tiller, are in reading state
+$(dirname "$BASH_SOURCE")/wait-pods-running-state.sh kube-system 
+
+# install all helm charts
+kubectl create ns ${NAMESPACE}
+linux-amd64/helm install --namespace=${NAMESPACE} $(pwd)/helm/prometheus-operator --name prometheus-operator
+linux-amd64/helm install --namespace=${NAMESPACE} $(pwd)/helm/kube-prometheus --name kube-prometheus
+
+# check if all pods are ready 
+$(dirname "$BASH_SOURCE")/wait-pods-running-state.sh ${NAMESPACE}
```
Comment (brancz) [inline helm/hack/e2e-test.sh]: This script doesn't exist yet

File: scripts/travis-e2e.sh [hunk 2]
```diff
@@ -36,3 +36,4 @@ kubectl apply -f scripts/minikube-rbac.yaml
 
 make crossbuild
 make e2e
+make e2e-helm
```
Comment (brancz) [inline scripts/travis-e2e.sh]: I think I'd prefer to have this in a separate job as this gives the notion, that the previous steps are related/required, when they're not

**Comment by gianrubio** [comment]
Took a loooong time to figure out how to run all this test on helm. I hope the kubernetes charts team provide the helm container image to test ASAP. 

I guess this e2e tests are acceptable for now, at least helm will not break after PR.

**Comment by brancz** [comment]
@gianrubio we're currently in the process of removing the migration(-tests) as they are not necessary anymore as TPRs have entirely been removed from the Kubernetes API in 1.8, so the failure you're seeing won't be occurring anymore.

**Comment by gianrubio** [comment]
I thought the error was caused by my changes, I'll wait until the changes are merged :)

**Comment by brancz** [comment]
Thanks for your patience! We fixed everything, you should be able to rebase and resolve the conflicts and everything should work. As we did a lot of changes I'll keep an eye on the failures here as well though. Also feel free to ping me if you think anything seems fishy.

**Comment by gianrubio** [comment]
Code is rebased and everything looks fine!

**Comment by brancz** [comment]
Nice! Thanks a lot! lgtm

**Pull request merged**